sequencer/src/api/sql.rs

```diff
@@ -333,9 +333,16 @@ async fn reconstruct_state<'a>(
     // Get the initial state.
     let mut parent = from_leaf;
     let mut state = ValidatedState::from_header(parent.block_header());
+
     // Pre-load the state with the accounts we care about to ensure they will be present in the
     // final state.
-    state.fee_merkle_tree = load_accounts(tx, from_height, accounts)
+    let mut accounts = accounts.iter().copied().collect::<HashSet<_>>();
+    // Add in all the accounts we will need to replay any of the headers, to ensure that we don't
+    // need to do catchup recursively.
+    accounts.insert(instance.chain_config.fee_recipient);
+    accounts.extend(fee_account_dependencies(instance, parent, &leaves).await);
```

Comment:
Does this change what parent the parent leaf is, and is later used in `compute_state_update`

Reply:
Nope, can't change `parent` if we're passing it in by value. Rust is helpful here